### ISSUE TITLE
Awesome work!

### ISSUE THREAD

**Issue by AlexIII** [ihrapsa/HFS-P3_DevKit#1]
Hi! Just wanted to say you did awesome job here. Thanks for that!
I had an idea to write custom software for this device so it wold be able to communicate with PC via USB (send measurements history and real-time). Sadly, according to the schematics USB connector does not have traces to the microcontroller (even though the MCU does supports USB). So this not gonna be possible with this hardware.

**Comment by ihrapsa** [comment]
Hi, and thanks for the interest! 

According to the datasheet this mcu does not support USB at all. This chip is the “N” version (FM33LC043N)

![image](https://github.com/user-attachments/assets/85ebf7e7-fa6c-4336-8c3a-b678ddead19d)



Based on my pcb tracing work, the USB C port has only a sensing GPIO (Pin11/PB8) that is wired to VBUS line over a voltage devider, as well as VCC in of the battery charging controller . I’m not sure how this can be hijacked and turned into a data out pin even with a custom breakout usbc plug board.

PIN11 = PB8/SEG7/SPI1_SSN/ATIM_CH3N


![image](https://github.com/user-attachments/assets/6dc78b00-e121-4076-8580-594f1151ae75)


Currently only way to extract history data from internal flash would be either over SWD port or some sort of hardware mod.


Thanks again for the interest and let me know if you have any kind of input or ideas 😁

**Comment by AlexIII** [comment]
> According to the datasheet this mcu does not support USB at all. This chip is the “N” version

Ah, I see, missed that. I suppose one other option would be to wire uart tx/rx to the unused typec pins and make external 'datacable' with generic usb-uart ttl converter. But yea, not the most elegant of solutions.

**Comment by ihrapsa** [comment]
Yeah, unfortunately the only broken out and useful UART gpios are the one used as SDA and SCL for OLED I2C bitbanging (which are already pulled up).

The alternative would be to use the SWD pads

**Comment by ihrapsa** [comment]
Another idea struck me:
I know some  operational electronic pocket dosimeters (like the Thermo Scientific EPD-G Mk2.5) have an IR led through which data can be exported over to an USB IR receiver.

HFS-P3 has two LEDs I could use to transfer data over “Li-Fi”
![image](https://github.com/user-attachments/assets/44508628-c58d-476f-b92f-83ae818a7b75)

This is an Arduino DIY PoC:
https://youtu.be/IdU6eCJ9Rh0

* A light receiver (either LDR or photodiode based) is needed
* The advantage is that no hardware mod of the device is required.

**Comment by AlexIII** [comment]
Wow, this it brilliant! The problem however is that it's one-way communication and will require some kind of makeshift fixture to secure the light receiver onto the device...

I too was thinking about using the LEDs, but rather make the LED pins pull double duty: switch their function to tx/rx when communicating with a PC. Alternatively, same can be done with the LCD pins SCL/SDA: we can pull down LCD reset for the duration of communication with a PC (at the price of having screen not working when the cable is plugged in).

**Comment by RChadwick7** [comment]
I like the idea of SWD lines on the USB-C port. This way, no matter what, we won't have to open anything to flash. Also, it won't require a special bootloader or code.
